src/A11yInterface.js

```diff
@@ -46,29 +62,44 @@ A11yInterface.propTypes = {
  */
 
 export default function A11yInterface(props) {
-  const { ariaLabelGenerator, numFrames, onKeyDown, height, width } = props;
+  const {
+    ariaLabelGenerator,
+    onKeyDown,
+    height,
+    width,
+    datasetWithAccessor,
+    xScale,
+  } = props;
+
+  const domain = xScale.domain();
+
+  // determine number of frames from n datasets with potentially different accessors
+  const mappedDatasets = datasetWithAccessor.map(({ data, accessor }) =>
+    data.map(d => getValue(accessor, d)),
+  );
+  const unionDatapoints = union(...mappedDatasets);
+  const numFrames = unionDatapoints.length;
   const sliceWidth = width / (numFrames - 1);
 
   return (
     <g className="rct-chart-a11y">
-      {[...Array(numFrames).keys()].map((_d, frameIndex) => (
+      {unionDatapoints.map((d, index) => (
         <rect
           className="rct-chart-visually-hidden-rect"
-          aria-label={ariaLabelGenerator(frameIndex)}
-          key={frameIndex}
+          aria-label={ariaLabelGenerator(d)}
```

Comment:
This variable name is a bit confusing because if I'm reading this correct it seems like d is really an `xValue`. Seems like `ariaLabelGenerator` will almost always want the array of all data points for that xValue so how about figuring that out here in order to simplify the handlers that are passed in? This line could be something like 
```
aria-label={ariaLabelGenerator(xValue, data, index)}
```
where data is the array of data just or that xValue, not the whole data for the chart.

You'll probably have to change line 78 to something like this:

```js
data.map(d => [d, getValue(accessor, d)]),
```

Reply:
My one concern is this approach could lose the context of the data. We can return the data points, but it could become unclear to the end user which data point belongs to which dataset it belongs to. For instance, I could have a schema like:

```
{
   name: 'Beyonce'
   data: [array of values to be rendered]
}

{
  name: 'Drake'
  data: [...]
}
```

In the schema I defined, that context (`name`) is lost when returning the two datapoints for a given `xValue`, and since aria-labels need to provide that context so the user understand what they're hearing, I'm not sure this is a better approach.

Reply:
Can you give a full example with that structure? All the examples just have an array of data with no name. Isn't the same end user who's defining `ariaLabelGenerator` also defining `datasetWithAccessor` and could reasonably assume that the data points come in the same order?

Reply:
The issue is less with the datapoint and more with needing context outside of the array used to generate the LineChart, which currently is all that's being used in `datasetWithAccessor`

Reply:
So in the example I provided, we'd only be passing in the `data` attribute to the `LineChart` and `datasetWithAccessor`, but we'd want to know what `name` data0 and data1 belong to in the `ariaLabelGenerator`. The user can define this themselves if we provide the `xValue`, but I worry about providing the other arguments if the end user doesn't actually get value given multiple datasets. They'd have to do that figuring out themselves anyway.

Reply:
Oh I see, my suggestion was to pass the `xValue` **and** the `data` (and even the index in case people need it) since I'm guessing in 99% of cases (and both your docs examples) users will want the data. So in your 2 datasets example, instead of 

```js
  const ariaLabelGenerator = xValue => {
    const data0Point = data0.find(d => d.x === xValue);
    const data1Point = data1.find(d => d.x === xValue);
    let ariaLabelString = `x Value, ${xValue}`;
    if(data0Point) {
      ariaLabelString += `, data0 y Value ${data0Point.y}`;
    }
    if(data1Point) {
      ariaLabelString += `, data1 y Value ${data1Point.y}`;
    }
    return ariaLabelString;
  }
```

you could have

```js
  const ariaLabelGenerator = (xValue, [data0Point, data1Point], /* index */) => {
    let ariaLabelString = `x Value, ${xValue}`;
    if(data0Point) {
      ariaLabelString += `, data0 y Value ${data0Point.y}`;
    }
    if(data1Point) {
      ariaLabelString += `, data1 y Value ${data1Point.y}`;
    }
    return ariaLabelString;
  }

Reply:
Sounds good, I'll make those changes this week and update the PR 👍